manifests/puppet.pp

```diff
@@ -43,11 +43,11 @@
     pubkey { $ssl_ca_cert:
       key_pair => $::certs::server_ca,
     } ~>
-    file { $client_key:
-      ensure => file,
-      owner  => '52',
-      mode   => '0400',
+    file { [$client_cert, $client_key, $ssl_ca_cert]:
+      ensure  => file,
+      owner   => 'puppet',
+      mode    => '0400',
+      require => Class['puppet::server::install'],
```

Comment:
@stbenjam do you think this would resolve http://projects.theforeman.org/issues/17863?  Or do you think its already resolved by https://github.com/Katello/puppet-certs/commit/67d027e7e00117cc8b51fc389be116e858ec268c